src/hyp3_isce2/burst.py

```diff
@@ -0,0 +1,339 @@
+    """Class that contains the parameters nessecary to request a burst from the API."""
+
+    granule: str
+    swath: str
+    polarization: str
+    burst_number: int
+
+
+class BurstMetadata:
+    """Class that contains the metadata for a burst."""
+
+    def __init__(self, metadata: ET.Element, burst_params: BurstParams):
+        self.safe_name = burst_params.granule
+        self.swath = burst_params.swath
+        self.polarization = burst_params.polarization
+        self.burst_number = burst_params.burst_number
+        self.manifest = metadata[0]
+        self.manifest_name = 'manifest.safe'
+        metadata = metadata[1]
+
+        names = [x.attrib['source_filename'] for x in metadata]
+        lengths = [len(x.split('-')) for x in names]
+        swaths = [x.split('-')[y - 8] for x, y in zip(names, lengths)]
+        products = [x.tag for x in metadata]
+        swaths_and_products = list(zip(swaths, products))
+
+        files = {'product': 'annotation', 'calibration': 'calibration', 'noise': 'noise'}
+        for name in files:
+            elem = metadata[swaths_and_products.index((self.swath.lower(), name))]
+            content = copy.deepcopy(elem.find('content'))
+            content.tag = 'product'
+            setattr(self, files[name], content)
+            setattr(self, f'{files[name]}_name', elem.attrib['source_filename'])
+
+        file_paths = [x.attrib['href'] for x in self.manifest.findall('.//fileLocation')]
+        pattern = f'^./measurement/s1.*{self.swath.lower()}.*{self.polarization.lower()}.*.tiff$'
+        self.measurement_name = [Path(x).name for x in file_paths if re.search(pattern, x)][0]
+
+        self.gcp_df = self.create_gcp_df()
+        self.footprint = self.create_geometry(self.gcp_df)[0]
+        self.orbit_direction = self.manifest.findtext('.//{*}pass').lower()
+
+    @staticmethod
+    def reformat_gcp(point: ET.Element) -> dict:
+        """Reformat a burst geolocation grid point to a dictionary.
+
+        Args:
+            point: The geolocation grid point to reformat.
+
+        Returns:
+            A dictionary containing the geolocation grid point's line, pixel, latitude, longitude, and height.
+        """
+        attribs = ['line', 'pixel', 'latitude', 'longitude', 'height']
+        values = {}
+        for attrib in attribs:
+            values[attrib] = float(point.find(attrib).text)
+        return values
+
+    def create_gcp_df(self) -> pd.DataFrame:
+        """Create a dataframe of geolocation grid points.
+
+        Returns:
+            A dataframe containing the geolocation grid points for the burst.
+        """
+        points = self.annotation.findall('.//{*}geolocationGridPoint')
+        gcp_df = pd.DataFrame([self.reformat_gcp(x) for x in points])
+        gcp_df = gcp_df.sort_values(['line', 'pixel']).reset_index(drop=True)
+        return gcp_df
+
+    def create_geometry(self, gcp_df: pd.DataFrame) -> Tuple[geometry.Polygon, Tuple[float], Tuple[float]]:
```

Comment:
OK, this function went over my head, but I think that's more due to my own lack of knowledge in this area rather than any fault of the code. So I left some feedback on the unit test instead.